src/classes/CW_AssociateAudienceRewardHelper.cls

```diff
@@ -0,0 +1,236 @@
+		newagsrset=new set<string>();
+		delagsrset=new set<Id>();  	
+		
+		//set the data display mode
+		agList=  [SELECT Id,Name FROM CW_Audience_Group__c  WHERE Campaign__c=:campId AND Is_Control_Group__c=false];
+		schList= [SELECT Id,Name ,Start_Date__c,End_Date__c  FROM CW_Schedule__c WHERE Marketing_Campaign__c=:campId ];
+		rewList= [SELECT Id,Name,Reward_Catalogue__r.Offer_Type__r.Name,Reward_Type__r.Name FROM CW_Campaign_Reward__c WHERE Campaign_Offer__r.Campaign__c=:campId ];
+		
+		for(CW_Audience_Group_Schedule__c rec: [SELECT Id,name,Audience_Group__c,Schedule__c FROM CW_Audience_Group_Schedule__c])
+		{
+			agsMapId.put(''+rec.Schedule__c+rec.Audience_Group__c,rec.Id);
+		}
+		
+		for(CW_Audience_Group_Schedule_Reward__c rec:[Select Id,name,Audience_Group_Schedule__r.Schedule__c,Audience_Group_Schedule__r.Audience_Group__c,Campaign_Reward__c FROM CW_Audience_Group_Schedule_Reward__c])
+		{  
+			agsrMap.put(''+rec.Audience_Group_Schedule__r.Schedule__c+rec.Audience_Group_Schedule__r.Audience_Group__c+rec.Campaign_Reward__c,rec.id);
+			agsSet.Add(''+rec.Audience_Group_Schedule__r.Audience_Group__c+rec.Campaign_Reward__c);
+		}
+	}
+ 
+	/*										*/
+	/*   SELECT & DESELECT FUNCTIONALITY  	*/
+	/*										*/
+
+	//handle item selected
+	private void doSelectGroupScheduleInternal(String item)
+	{
+		newagsset.Add(item);
+	}
+	
+	private void doDeSelectGroupScheduleInternal(String item)
+	{
+		newagsset.remove(item);
+	}
+	
+	public void doSelectGroupSchedule()
+	{
+		doSelectGroupScheduleInternal(this.contextItem);
+
+		String GrpId=this.contextItem;
+		for(CW_Schedule__c shr:schList)
+		{
+			doSelectGroupRewardInternal(''+shr.Id+GrpId);
+		}
+	}
+
+	//handle item deselected
+	public void doDeSelectGroupSchedule()
+	{
+		doDeSelectGroupScheduleInternal(this.contextItem);	
+		String GrpId=this.contextItem;
+		for(CW_Schedule__c shr:schList)
+		{
+				doDeSelectGroupRewardInternal(''+shr.Id+GrpId);
+		}
+	}
+	
+	private void doSelectGroupRewardInternal(String item)
+	{
+		id agsrId= agsrMap.get(item);
+		if(agsrId!=null)
+			// rec uncheked - remove from delete
+			delagsrset.Remove(agsrId);
+		else
+			newagsrset.Add(item);
+	}
+
+	//handle item deselected
+	private void doDeSelectGroupRewardInternal(String item)
+	{
+		id agsrId= agsrMap.get(Item);
+		if(agsrId!=null)
+			delagsrset.Add(agsrId);
+		else
+			newagsrset.remove(Item);
+	}
+	
+	
+	public void doSelectGroupReward()
+	{
+		doSelectGroupRewardInternal(this.contextItem);
+		doSelectGroupScheduleInternal(this.contextItem.substring(18));
+	}
+
+	//handle item deselected
+	public void doDeSelectGroupReward()
+	{
+		doDeSelectGroupRewardInternal(this.contextItem);
+	}
+	
+	private void deleteSetOfIds(Set<id> toDel,string sType){
+		List<sobject> lst= new List<sobject>();
+		
+		for(Id recid:toDel){
+			Sobject delobj= Schema.getGlobalDescribe().get(sType).newSObject();
+			delobj.Id=recid;
+			lst.Add( delobj);
+		}
+		delete lst;
+	}
+	
+	public void PresistChages()
```

Comment:
Style issue: methods should use the "lowerCamelCase" notation. Suggestion: rename accordingly.